generated main does not compile on OSX

I got the following compile error for an orogen generated project:

big_mockup/.orogen/main-orogen_default_big_mockup__Workaround.cpp:286:20: error: 
      no member named 'sa_restorer' in 'sigaction'
    sigint_handler.sa_restorer  = 0;
    ~~~~~~~~~~~~~~ ^
1 warning and 1 error generated.

The Linux manpage (http://man7.org/linux/man-pages/man2/sigaction.2.html) mentions the following:

The sa_restorer field is not intended for application use.  (POSIX
       does not specify a sa_restorer field.)  Some further details of
       purpose of this field can be found in sigreturn(2).

It's probable safe to remove this line?

--- a/lib/orogen/templates/main.cpp
+++ b/lib/orogen/templates/main.cpp
@@ -70,6 +70,8 @@ bool exiting;
 QApplication *qapp;
 <% end %>
 
+#include <string.h>
+
 namespace orogen
 {
 <% task_activities.each do |task| %>
@@ -402,10 +404,9 @@ RTT::internal::GlobalEngine::Instance(ORO_SCHED_OTHER, RTT::os::LowestPriority);
     }
 
     struct sigaction sigint_handler;
+    memset(&sigint_handler, 0, sizeof(sigint_handler));
     sigint_handler.sa_handler = &sigint_quit_orb;
     sigemptyset(&sigint_handler.sa_mask);
-    sigint_handler.sa_flags     = 0;
-    sigint_handler.sa_restorer  = 0;
     if (-1 == sigaction(SIGINT, &sigint_handler, 0))
     {
         std::cerr << "failed to install SIGINT handler" << std::endl;

